### ISSUE TITLE
KeyError: 'interval'

### ISSUE THREAD

**Issue by nicx** [gcobb321/icloud3#9]
I am using the latest iCloud3.py with HA 0.86.1 and get this error on start:

```
2019-01-24 11:35:17 ERROR (MainThread) [homeassistant.core] Error doing job: Future exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/device_tracker/icloud3.py", line 845, in _device_polling_15_sec_timer_loop
    self._update_tracked_devices()
  File "/config/custom_components/device_tracker/icloud3.py", line 1040, in _update_tracked_devices
    attrs[CONF_INTERVAL],
KeyError: 'interval'
```

after this it seems some data is fetching, but I get regularly this error:

```
2019-01-24 11:35:46 ERROR (SyncWorker_6) [custom_components.device_tracker.icloud3] ►►Error on retrieving data: Internal Error
2019-01-24 11:35:52 ERROR (SyncWorker_6) [custom_components.device_tracker.icloud3] ►►Error on retrieving data: Internal Error
2019-01-24 11:36:16 ERROR (SyncWorker_5) [custom_components.device_tracker.icloud3] ►►Error on retrieving data: Internal Error
2019-01-24 11:36:22 ERROR (SyncWorker_5) [custom_components.device_tracker.icloud3] ►►Error on retrieving data: Internal Error

```

what can I do? :)

**Comment by gcobb321** [comment]
The error is on a statement writing information to the log file that I can not reproduce. I have completely changed the way it now writes the log records and hopefully, that will catch the error. Or at least see what is wrong. I would check the config file and remove any 'interval' item.  I've also made a few other changes (added 'ignore_gps_accuracy_inzone' configuration variable (it defaults to true) that was creating other problems, cleaned up some code and added a toggle so debug info is written to the log file without making changes to the log parameters in the configuration file and restarting HA.

Send your email address to geekstergary@gmail.com and I'll send you a prerelease version of iCloud3 v.87, along with a script that will toggle writing a lot of debug messages to the log file.

**Comment by gcobb321** [comment]
Fixed in 0.86.1.

**Comment by zyjex** [comment]
i've tried your plugin for a few days now, but i gotten this error 2 days in a row, it seems like it happens when i leave home or when i arrive home. It crashes home assistant and makes it unresponsive.

2019-01-31 15:28:31 ERROR (MainThread) [homeassistant.core] Error doing job: Future exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/device_tracker/icloud3.py", line 845, in _device_polling_15_sec_timer_loop
    self._update_tracked_devices()
  File "/config/custom_components/device_tracker/icloud3.py", line 1040, in _update_tracked_devices
    attrs[CONF_INTERVAL],
KeyError: 'interval'

and then home assistant cant be reached, and its just sits like this forever

2019-01-31 15:33:40 WARNING (MainThread) [homeassistant.helpers.entity] Update of switch.farstu_ute_311 is taking over 10 seconds
2019-01-31 15:33:41 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.processor_use is taking over 10 seconds
2019-01-31 15:33:41 WARNING (MainThread) [homeassistant.helpers.entity] Update of lock.idlock_575 is taking over 10 seconds
2019-01-31 15:33:41 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.ackumulatortank_382 is taking over 10 seconds
2019-01-31 15:33:47 WARNING (MainThread) [homeassistant.helpers.entity] Update of binary_sensor.kameraping is taking over 10 seconds
2019-01-31 15:33:51 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.josefs_iphone_battery_state_2 is taking over 10 seconds
